client/app/scripts/liveblog-edit/posts.service.js

```diff
@@ -123,47 +123,64 @@ define([
-            angular.extend(post, {
-                // add a `multiple_items` field. Can be false or a positive integer.
-                // FIXME: left like that to support other feature, but this need to be in camelcase
-                multiple_items: post.groups[1].refs.length > 1 ? post.groups[1].refs.length : false,
-                // add a `mainItem` field containing the first item
-                mainItem: post.groups[1].refs[0],
-                items: post.groups[1].refs
-            });
-            // if an item has a commenter then that post hasComments.
-            post.hasComments = _.reduce(post.groups[1].refs, function(is, val) {
-                return is || !_.isUndefined(val.item.commenter);
-            }, false);
-            // `fullDetails` is a business logic that can be compiled from other objects.
-            post.fullDetails = post.hasComments;
-            // special cases for comments.
-            post.showUpdate = (post.content_updated_date !== post.published_date) &&
-                               (!post.hasComments) && (post.mainItem.item.item_type !== 'comment');
-            angular.forEach(post.items, function(val) {
-                if (post.fullDetails) {
-                    _completeUser(val.item);
+            return $q(function(resolve, reject) {
+                angular.extend(post, {
+                    // add a `multiple_items` field. Can be false or a positive integer.
+                    // FIXME: left like that to support other feature, but this need to be in camelcase
+                    multiple_items: post.groups[1].refs.length > 1 ? post.groups[1].refs.length : false,
+                    // add a `mainItem` field containing the first item
+                    mainItem: post.groups[1].refs[0],
+                    items: post.groups[1].refs
+                });
+
+                //console.log('complete post', pos.particular_type);
+
+                // if an item has a commenter then that post hasComments.
+                post.hasComments = _.reduce(post.groups[1].refs, function(is, val) {
+                    return is || !_.isUndefined(val.item.commenter);
+                }, false);
+                // `fullDetails` is a business logic that can be compiled from other objects.
+                post.fullDetails = post.hasComments;
+                // special cases for comments.
+                post.showUpdate = (post.content_updated_date !== post.published_date) &&
+                                   (!post.hasComments) && (post.mainItem.item.item_type !== 'comment');
+                angular.forEach(post.items, function(val) {
+                    if (post.fullDetails) {
+                        _completeUser(val.item);
+                    }
+                });
+                _completeUser(post.mainItem.item);
+
+                if (post.syndication_in) {
+                    api.syndicationIn.getById(post.syndication_in).then(function(synd) {
```

Comment:
Can we not cache the blog title somehow? Not sure if we need to worry about it, but this seems to be a pretty sub-optimal way of doing things. Alternative would be to do the preprocessing on the server, but you could just add an optimisation task in Jira for this.